### PR TITLE
#163426500 View all user bookmarked article

### PR DESCRIPTION
**Pull request by biolabalo**
#### What does this PR do?
* It creates a page where a viewer can view all the articles they have written.
* it adds the delete button to delete  articles a user created

#### What are the relevant pivotal tracker stories?
[#163426500](https://www.pivotaltracker.com/story/show/163426500)
#### Screenshots (if appropriate)

image
image
image

### CODE REVIEW

**Review by EfeAgare — DISMISSED** [review]

**Review by ozone4real — DISMISSED** [review]
LGTM

File: src/components/UserBookmarksPage.jsx [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+          id={number}
+          onClick={this.handleClick}
+        >
+          {number}
+        </li>
+    ));
+
+    return (
+      <Fragment>
+        {isLoading ? <TopBarProgress /> : '' }
```
Comment (p-funky) [inline src/components/UserBookmarksPage.jsx]: Short-circuit preffered.
```suggestion
        {isLoading && <TopBarProgress />}
```

File: src/components/UserBookmarksPage.jsx [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    if (isVerified) this.props.getAllUsersBookMarkedArticles();
+  }
+
+  render() {
+    let bookmarkView;
+
+    const { currentPage, bookmarksPerPage } = this.state;
+    const { bookmarkedArticles: { isLoading, userBookmarks } } = this.props;
+
+    if (!isLoading && userBookmarks.length === 0) {
```
Comment (p-funky) [inline src/components/UserBookmarksPage.jsx]: Let's take advantage of javascript's falsy values
```suggestion
    if (!isLoading && !userBookmarks.length) {
```

File: src/components/UserBookmarksPage.jsx [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+      bookmarkView = <h1> You Currently do not have any bookmarked article   </h1>;
+    }
+    if (!isLoading && userBookmarks.length !== 0) {
```
Comment (p-funky) [inline src/components/UserBookmarksPage.jsx]: Let's take advantage of javascript's truthy values
```suggestion
    if (!isLoading && userBookmarks.length) {
```

File: src/components/common/UserArticleCardComponent.js [hunk 4]
```diff
@@ -19,73 +32,76 @@ const UserArticleCard = ({
...
+</Button>
+</ButtonGroup>
+)}
+</div>
+</div>
+</div>
+</Card.Body>
+</Card>
+</Fragment>
   );
```
Comment (p-funky) [inline src/components/common/UserArticleCardComponent.js]: Why did the formatting here disappear? Do you have a reverse auto formatter? All the indentation seemed to disappear.

File: src/components/common/UserArticleCardComponent.js [hunk 5]
```diff
@@ -1,14 +1,27 @@
 import React, { Fragment } from 'react';
 import { NavLink } from 'react-router-dom';
+import swal from 'sweetalert';
```
Comment (p-funky) [inline src/components/common/UserArticleCardComponent.js:3]: I thought we were no longer using sweetalert?

Comment (biolabalo) [reply src/components/common/UserArticleCardComponent.js:3]: I am using it to display a warning modal before you delete an article. Toast does not have a modal

File: src/components/common/UserArticleCardComponent.js (hunk 4 above)
Comment (biolabalo) [reply src/components/common/UserArticleCardComponent.js]: I just noticed it when the pr was raised

**Review by divinediscipline — DISMISSED** [review]
Looks good to me

**Review by ozone4real — DISMISSED** [review]
LGTM

**Pull request merged**